dotnet/src/Azure.Iot.Operations.Connector/Models/DeviceIdentity.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Azure.Iot.Operations.Connector.Models;
+
+/// <summary>
+/// Represents device identity information used for CloudEvents correlation.
+/// This class captures all device-level identifiers needed to construct
+/// CloudEvents headers according to the message_correlation.md specification.
+/// </summary>
+public class DeviceIdentity
```

Comment:
Most of this information is already present in the ADR model type for a device ([here](https://github.com/Azure/iot-operations-sdks/blob/main/dotnet/src/Azure.Iot.Operations.Services/AssetAndDeviceRegistry/Models/Device.cs)).

From what I understood from the email thread, the compound key is the only new addition here, but Alex is proposing it to be modeled in the ADR service.

With that in mind, I'm not sure that we would need this connector-level representation of a device